### ISSUE TITLE
rebalancing: Rebalancing criteria is not effective for nodes running on the same machine

### ISSUE THREAD

**Issue by mrtracy** [cockroachdb/cockroach#2600]
A common test scenario is to run multiple nodes on the same machine. However, the current rebalancing criteria is not a good match for this scenario; this is because stores are compared based on "Capacity.FractionUsed()", which is based on information about the hard-drive the store is located on.  In the case of multiple nodes on the same machine, they are often running from different files on the same hard-drive, and thus the FractionUsed() is always the same for all nodes.

We likely need an alternative rebalancing criteria for this situation, which will be very common in tests.

**Comment by tamird** [comment]
s/machine/store/, right? Also, do you actually want a rebalance to occur in this case?

**Comment by bdarnell** [comment]
The issue is when multiple stores are backed by the same physical filesystem. This is a test-only scenario, but for testing we need some way to either apply different limits to each node (can we do this with docker?) or a way to fake the "available space on filesystem" functions.

**Comment by jacqueswww** [comment]
@bdarnell try creating a new docker container with added parameter "--storage-opt dm.basesize=20G":
```
docker run -t -i -p 8080:8080 cockroachdb/cockroach shell --storage-opt dm.basesize=20G
```

**Comment by jacqueswww** [comment]
@bdarnell try creating a new docker container with added parameter "--storage-opt dm.basesize=20G":

```
docker --storage-opt dm.basesize=10G pull cockroachdb/cockroach
```

Docker daemon needs to switch to device mapper storage though:

DOCKER_OPTS="--storage-driver=devicemapper"

(http://muehe.org/posts/switching-docker-from-aufs-to-devicemapper/)

**Comment by mrtracy** [comment]
This was fixed by #3098, which added the `--balance-mode` flag.  

Running cockroach with `--balance-mode=rangecount` will fix this problem.
